src/praisonai-agents/tests/unit/test_ingress_admission.py

```
@@ -0,0 +1,176 @@
+        is_command=False,
+    )
+    assert d.admit is True
+    assert d.gate == GATE_DIRECT
+
+
+def test_decision_is_frozen():
+    d = resolve_ingress_admission(chat_type="dm", sender_id="u1")
+    assert isinstance(d, IngressDecision)
+    try:
+        d.admit = False  # type: ignore[misc]
+    except Exception as exc:  # frozen dataclass raises FrozenInstanceError
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)